infer/src/base/ProcessPool.ml

```diff
@@ -446,6 +462,56 @@ module Worker = struct
   let id_to_int = Fn.id
 end
 
+(** Data marshalled to describe what a child spawned by [spawn_child] below must do. *)
+type ('work, 'result, 'final) child_data =
+  {slot: int; child_prologue: unit -> unit; f: 'work -> 'result option; epilogue: unit -> 'final}
+
+(** Spawn a new child and start it so that it is ready for work.
+
+    The child inherits [updates_w] to send updates up to the parent, and a new pipe is set up for
+    the parent to send instructions down to the child. *)
+let spawn_child ~child_prologue ~slot (_updates_r, updates_w) ~f ~epilogue =
+  let to_child_r, to_child_w = Unix.pipe () in
+  let orig_args = Sys.get_argv () in
+  let args =
+    (* Craft the command-line for the subprocesses: add --run-as-child <n> to the current one *)
+    match Array.to_list orig_args with
+    | [] ->
+        assert false (* always contain the name of the binary *)
+    | hd_args :: tl_args ->
+        let new_argl = hd_args :: "--run-as-child" :: string_of_int slot :: tl_args in
+        Array.of_list new_argl
+  in
```

Comment:
alternatively you can pass arguments via `INFER_ARGS`, which is what we do to pass arguments to infer sub-processes started by build systems, for instance. This could just be `CommandLineOption.extend_env_args ["--run-as-child"; string_of_int slot] ;`.

Reply:
We did not do that initially, because having the "identity" of the child visible  in the command-line _could_ be useful. However:
- it's not easily accessible in fork mode
- we've used this functionality once in 3 years, IIRC.
So I think this change is fine. (And you've done it in a later commit.)